content/en/blog/_posts/Use containerd to handle k8s.gcr.io deprecation

```diff
@@ -0,0 +1,174 @@
+
+Turns out you can use containerd to handle this redirection at node level. Let’s find out how.
+
+## Using mirrors in containerd
+
+Ever since Dockerhub started rate limiting image pulls, many have opted to store images in local registries. Mirrors save network bandwidth, reduce image pull time, and don’t rate-limit.
+
+You can configure `registry.k8s.io` as a mirror to `k8s.gcr.io` in containerd. This configuration will **automatically pull images from `registry.k8s.io`** whenever a Pod uses an image stored in `k8s.gcr.io`.
+
+On your worker node, append these lines in the containerd config file at `/etc/containerd/config.toml`:
+
+```bash
+[plugins."io.containerd.grpc.v1.cri".registry]
+   config_path = "/etc/containerd/certs.d"
+```
+
+The final file on an Amazon EKS cluster looks like this:
+
+```yaml
+version = 2
+root = "/var/lib/containerd"
+state = "/run/containerd"
+
+[grpc]
+address = "/run/containerd/containerd.sock"
+
+[plugins."io.containerd.grpc.v1.cri".containerd]
+default_runtime_name = "runc"
+
+[plugins."io.containerd.grpc.v1.cri"]
+sandbox_image = "602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/pause:3.5"
+
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
+runtime_type = "io.containerd.runc.v2"
+
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
+SystemdCgroup = true
+
+[plugins."io.containerd.grpc.v1.cri".cni]
+bin_dir = "/opt/cni/bin"
+conf_dir = "/etc/cni/net.d"
+
+[plugins."io.containerd.grpc.v1.cri".registry]
+config_path = "/etc/containerd/certs.d"
+```
+
+Next, create a directory called `k8s.gcr.io` and a `hosts.toml` file inside it:
+
+```bash
+mkdir -p /etc/containerd/certs.d/k8s.gcr.io
+
+cat << EOF > /etc/containerd/certs.d/k8s.gcr.io/hosts.toml
+server = "https://k8s.gcr.io"
+
+[host."https://registry.k8s.io"]
+capabilities = ["pull", "resolve"]
+EOF
+```
+
+Image pull requests to `k8s.gcr.io` will now be sent to `registry.k8s.io`.
+
+Restart containerd and kubelet for the change to take effect.
+
+```bash
+systemctl restart containerd kubelet
+```
+
+Let’s validate that images are indeed getting pulled from the new registry. I added an entry to my `/etc/hosts` file to break `k8s.gcr.io`:
+
+![Image.png](https://res.craft.do/user/full/9d54cc03-adfe-f72f-3389-565eb7356d1d/doc/13FE78CC-1DDD-47D9-B875-EE3FB15F916E/8AEBA792-CA42-4C8B-9C8D-E34AC55A9847_2/7A4cAKtUBNGVqmUysXRwpW0WGy3cNGz14xxujKlsmc8z/Image.png)
+
+Conatinerd can no longer pull an image from `k8s.gcr.io`
+
+![Image.jpeg](https://res.craft.do/user/full/9d54cc03-adfe-f72f-3389-565eb7356d1d/doc/13FE78CC-1DDD-47D9-B875-EE3FB15F916E/D382C5B0-437D-44E3-BAA2-E32032FF7E29_2/zmbCw5wl0LyV1TNJdiEQxCnSqKBS3s4IDZWYljKaLvkz/Image.jpeg)
+
+I can tell `ctr` to use the mirror by specifying the `—hosts-dir` parameter:
+
+```bash
+ctr images pull --hosts-dir "/etc/containerd/certs.d" k8s.gcr.io/pause:latest
```

Comment:
I thought the example was using a `602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/pause` as its sandbox image?

Reply:
The pause container I used is unrelated to the pause container EKS uses. Anyway, I have updated the steps to fetch kube-proxy instead of a pause container.